File: yyong/2025.05/250504_겹치는_건_싫어.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");

const [N, M] = input[0].split(" ").map(Number);
const sequence = input[1].split(" ").map(Number);
const numberCount = new Map();

let left = 0;

numberCount.set(sequence[left], 1);

let result = 1;

// 수열 범위 체크
for (let right = 1; right < N; right++) {
  const rightNum = sequence[right];
  numberCount.set(rightNum, (numberCount.get(rightNum) || 0) + 1);

  // 조건 맞을 때 까지 범위 조절
  while (numberCount.get(rightNum) > M) {
    const leftNum = sequence[left++];
    numberCount.set(leftNum, numberCount.get(leftNum) - 1);
  }

  result = Math.max(result, right - left + 1);
}

console.log(result);

File: min/2025/2025.05/250504_겹치는 건 싫어.py
from collections import defaultdict

N, K = map(int, input().split())
arr = list(map(int, input().split()))

cnt = defaultdict(int)

left = 0
max_len = 0
for right in range(N):
    cnt[arr[right]] += 1

    while cnt[arr[right]] > K:
        cnt[arr[left]] -= 1
        left += 1

    max_len = max(max_len, right - left + 1)

print(max_len)